addons/test_mail/tests/test_mail_activity.py

```diff
@@ -717,6 +722,232 @@ def test_my_activity_flow_employee(self):
             self.assertEqual(test_record_1, record)
 
 
+@tests.tagged('mail_activity')
+class TestMailActivity(TestActivityCommon):
```

Comment:
Same here, according to me: simplification required ! Avoid making "too complete" tests, because I have to say I would probably not be able to update this test if something fails of changes. According to me, you should keep tests simple, two IDs, one type, one variation with "keep done"; avoid multi employee (not sure what is adds). Don't re-do tests, having one past and one future activity is sufficien tfor example as we consider state is already tested, ...

We globally need one test ensuring we have most of the code covered; here it seems to me we are adding a lot of scheduling and activities to be sure to cover everything, but with a lot of redundant code...

KISS (keep it super simple) !